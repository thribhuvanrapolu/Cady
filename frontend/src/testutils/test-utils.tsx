import { Provider } from "react-redux";
import { render as rtlRender, RenderResult } from "@testing-library/react";
import { createStore } from "redux";
import { ReactElement, JSXElementConstructor, ReactNode } from "react";

function render(
  ui: ReactElement<any, string | JSXElementConstructor<any>>,
  {
    initialState = {},
    store = createStore((a) => a, initialState),
    ...renderOptions
  }: any = {}
): RenderResult {
  function Wrapper({ children }: { children: ReactNode }): any {
    return <Provider store={store}>{children}</Provider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";

export { render };
