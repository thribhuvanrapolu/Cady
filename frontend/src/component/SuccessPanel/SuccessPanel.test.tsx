import { screen } from "@testing-library/react";
import { SuccessPanel } from "./SuccessPanel";
import "@testing-library/jest-dom";
import { render } from "../../testutils/test-utils";

jest.mock("../SnackBar", () => ({
  useSnackBar: jest.fn().mockReturnValue(jest.fn()),
}));

jest.mock("../UserCalendar");

jest.mock("next/router", () => ({
  useRouter: jest.fn().mockReturnValue({ push: jest.fn() }),
}));

describe("SuccessPanel Component", () => {
  test("should match snapshot", () => {
    const { baseElement } = render(<SuccessPanel url="testing-my-url" />);
    expect(baseElement).toMatchSnapshot();
  });
  test("should render url", () => {
    render(<SuccessPanel url="testing-my-url" />);
    expect(screen.getByTestId("success-panel-url")).toHaveTextContent(
      "testing-my-url"
    );
  });
});
