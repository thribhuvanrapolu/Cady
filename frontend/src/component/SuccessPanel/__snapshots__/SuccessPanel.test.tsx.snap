// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPanel Component should match snapshot 1`] = `
<body>
  <div>
    <div
      class="w-full css-1p5q5e5-MuiStack-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-17qww7q-MuiTypography-root"
      >
        Events created successfully, please share the link below to your friends
      </h5>
      <div
        class="relative rounded-lg bg-gray-200"
      >
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButtonBase-root  css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ContentCopyIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
              />
            </svg>
          </span>
          Copy
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1j7gnzl-MuiTypography-root"
          data-testid="success-panel-url"
        >
          testing-my-url
        </p>
      </div>
    </div>
  </div>
</body>
`;
